src/deepsparse/loggers/build_logger.py

```diff
@@ -325,6 +332,83 @@ def possibly_modify_target_identifiers(
     return data_logging_config
 
 
+def parse_out_predefined_metric_function(
+    data_logging_config: Dict[str, List[MetricFunctionConfig]]
+) -> Dict[str, List[MetricFunctionConfig]]:
+    """
+    Parse out the predefined metric functions from the
+    data logging config and update the data_logging_config
+    accordingly
+
+    :param data_logging_config: The configuration of the data loggers
+    :return: The updated configuration of the data loggers
+    """
+    for target_identifier, metric_functions in data_logging_config.copy().items():
```

Comment:
hm, not sure about this copy-them-modify flow in terms of maintainability, especially since we are modifying a user input in-place. was this added in to address a dict mutation during iteration issue?
i'd consider either first parsing out any target identifiers for predefined, then iterating through those:
```python
predefined_identifiers = [identifier for identifier in data_logging_config if idenfier.startswith(...)]
for predefined_identifier in predefined_identifiers:
   # add and delete
```

or build a new data_logging_config and update/build on every loop iteration
```python
for target_identifier, metric_functions in data_logging_config.items():
    if target_identifier.endswith(PREDEFINED_IDENTIFIER_ALIAS):
        ...
    else:
       updated_data_logging_config[target_identifier] = metric_functions
```